test/Microsoft.Azure.WebJobs.Host.UnitTests/JobHostMetadataProviderTests.cs

```diff
@@ -174,19 +174,19 @@ public void DefaultTypeForOpenTypeTrigger()
         {
             var ext = new OpenTypeTriggerExtension();
             var host = new HostBuilder()
-                 .ConfigureDefaultTestHost(b =>
-                 {
-                     b.AddExtension(ext);
-                 })
-                 .ConfigureTypeLocator() // empty 
-                 .Build();
+                        .ConfigureDefaultTestHost(b =>
+                        {
+                            b.AddExtension(ext);
+                        })
+                        .ConfigureTypeLocator() // empty
+                        .Build();
             IJobHostMetadataProvider metadataProvider = host.CreateMetadataProvider();
 
             var attr = new Test9Attribute(null);
             var type = metadataProvider.GetDefaultType(attr, FileAccess.Write, null);
 
-            // The trigger handles Open type, which means it will first pull byte[]. 
-            Assert.Equal(typeof(byte[]), type);
+            // The trigger handles Open type, which means it will first pull ParameterBindingData.
+            Assert.Equal(typeof(ParameterBindingData), type);
```

Comment:
Wondering if this indicates a potential breaking change? Does ParameterBindingData need to come first in the precedence list? Wondering what potential impact this change will have.

Reply:
Yeah I am worried about this too, and maybe this is something you can help me understand better.

Without ParameterBindingData being at the top of the list, we never end up with the ParameterBindingData converter which ends up breaking the entire SDK-binding scenario. I did test out the other extensions that don't support ParameterBindingData and things were working as expected, so I haven't noticed anything breaking yet - but that doesn't mean that nothing will break 🥲

Do you happen to know why this was the original order of things? And bytes has to be first, do you have suggestions on how we can get the correct converter in the ParameterBindingData scenario?